connaisseur/validators/notaryv1/notaryv1_validator.py

```diff
@@ -240,16 +256,31 @@ def __search_image_targets_for_tag(trust_data: dict, image: Image):
     def __update_with_delegation_trust_data(
         self, trust_data, delegations, key_store, image
     ):
-        for delegation in delegations:
-            delegation_trust_data = self.notary.get_delegation_trust_data(
-                image, TUFRole(delegation)
-            )
-            # when delegations are added to the repository, but weren't yet used for
-            # signing, the delegation files don't exist yet and are `None`. in this
-            # case validation must be skipped
-            if delegation_trust_data is not None:
-                delegation_trust_data.validate(key_store)
-            trust_data[delegation] = delegation_trust_data
+        with Pool() as pool:
+            try:
+                result = pool.starmap_async(
+                    self.notary.get_delegation_trust_data,
+                    [(image, TUFRole(delegation)) for delegation in delegations],
+                )
+                delegation_trust_data_list = result.get(timeout=30)
+            except Exception as err:
+                msg = "Error retrieving delegation trust data form notary."
+                raise NotFoundException(
+                    message=msg, notary=str(self.notary.name)
+                ) from err
+
+        # when delegations are added to the repository, but weren't yet used for signing,
```

Comment:
This comment doesn't make sense anymore since TrustData() cannot be None if I see this correctly. However, this was already that way before this commit, so you probably changed that during your refactoring/modularization commit. Not having looked into it, it looks like we lost a case here? Can you investigate?